lib/index.js

```diff
@@ -1,28 +1,34 @@
 'use strict';
 
 var fs      = require('fs');
-var Promise = require('bluebird');
-var popen   = Promise.promisify(fs.open);
-var pclose  = Promise.promisify(fs.close);
+var util    = require('util');
+var popen   = util.promisify(fs.open);
+var pclose  = util.promisify(fs.close);
 var detect  = require('./detect');
 
-function measure (plugins, path, callback) {
+async function measure (plugins, path, callback) {
   var fileDescriptor;
 
-  return popen(path, 'r')
-  .then(function (fd) {
+  try {
+    const fd = await popen(path, 'r');
     fileDescriptor = fd;
-    return detect(fd, plugins);
-  })
-  .then(function (plugin) {
-    return plugin.measure(path, fileDescriptor);
-  })
-  .finally(function () {
+    const plugin = await detect(fd, plugins);
+
+    if (callback) return callback(undefined, await plugin.measure(path, fileDescriptor));
+
+    const measurement = await plugin.measure(path, fileDescriptor);
+    return measurement;
+  }
+  catch (err) {
+    if (callback) return err;
```

Comment:
Did you mean `if (!callback) throw err;` here?

Reply:
Ah yea good catch